packages/trpc/server/routers/viewer/availability/schedule/getAllSchedulesByUserId.handler.ts

```diff
@@ -0,0 +1,64 @@
+import { transformAvailabilityForAtom } from "@calcom/lib";
+import { hasReadPermissionsForUserId } from "@calcom/lib/hasEditPermissionForUser";
+import { prisma } from "@calcom/prisma";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../../trpc";
+import { getDefaultScheduleId } from "../util";
+import type { TGetAllByUserIdInputSchema } from "./getAllSchedulesByUserId.schema";
+
+type GetOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TGetAllByUserIdInputSchema;
+};
+
+export const getAllSchedulesByUserIdHandler = async ({ ctx, input }: GetOptions) => {
+  const { user } = ctx;
+
+  const schedules = await prisma.schedule.findMany({
+    where: {
+      userId: input.userId,
+    },
+    select: {
+      id: true,
+      userId: true,
+      name: true,
+      availability: true,
+      timeZone: true,
+    },
+  });
+
+  if (!schedules) {
+    throw new TRPCError({
+      code: "UNAUTHORIZED",
+    });
+  }
+  const isCurrentUserPartOfTeam = hasReadPermissionsForUserId({
+    ctx,
+    input: { memberId: input?.userId },
+  });
+
+  const isCurrentUserOwner = input?.userId === user.id;
+
+  if (!isCurrentUserPartOfTeam && !isCurrentUserOwner) {
+    throw new TRPCError({
+      code: "UNAUTHORIZED",
```

Comment:
Check for user role before querying the database

Reply:
Here the checks  `if(!isCurrentUserPartOfTeam && !isCurrentUserOwner)` , ensures that data (i.e. availabilities) of current user's or his team member's only can be queried. It should be possible regardless of the role.

Reply:
I meant do this check `if(!isCurrentUserPartOfTeam && !isCurrentUserOwner)`  before querying the database for schedules

Reply:
oh, ok! done